### PR TITLE
IA-1746 add api doc

### PR DESCRIPTION
**Pull request by Qi77Qi**
Use https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#composition-and-inheritance-polymorphism...

Tested on fiab

Screen Shot 2020-03-16 at 11 27 09 AM

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

File: http/src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -11,12 +11,14 @@ info:
 tags:
   - name: cluster
     description: Clusters API
+  - name: runtime
+    description: Runtime API. Support both Dataproc cluster and Gce instances.
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: Nit: maybe write out Google Compute Engine

File: http/src/main/resources/swagger/api-docs.yaml [hunk 2]
```diff
@@ -61,6 +63,62 @@ paths:
               schema:
                 $ref: "#/components/schemas/LeonardoVersion"
       security: []
+  /api/google/v1/runtime/{googleProject}/{name}:
+    post:
+      summary: Creates a new GCE compute instance in the given project with the given name.
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: Nit: GCE instance or Google Compute Engine instance

Comment (rtitle) [reply http/src/main/resources/swagger/api-docs.yaml]: Also this endpoint supports GCE or Dataproc

File: http/src/main/resources/swagger/api-docs.yaml [hunk 3]
```diff
@@ -61,6 +63,62 @@ paths:
...
+      description: >
+        * The request is completed without waiting for the initiation of the
+        cluster's creation on the Google side.
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: s/cluster/runtime

File: http/src/main/resources/swagger/api-docs.yaml [hunk 4]
```diff
@@ -61,6 +63,62 @@ paths:
...
+      parameters:
+        - in: path
+          name: googleProject
+          description: googleProject
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: name
+          description: runtime name. only lowercase alphanumeric characters, numbers and
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: maybe just
```
runtime name. only lowercase alphanumeric characters, numbers and dashes are allowed.
```

File: http/src/main/resources/swagger/api-docs.yaml [hunk 5]
```diff
@@ -61,6 +63,62 @@ paths:
...
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CreateRuntimeRequest"
+      responses:
+        "202":
+          description: Cluster creation request accepted
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: Runtime creation

File: http/src/main/resources/swagger/api-docs.yaml [hunk 6]
```diff
@@ -61,6 +63,62 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+      security:
+        - googleoauth:
+            - openid
+            - email
+            - profile
   /api/clusters:
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml:121]: Can we make `/api/google/v1/runtime…` appear in a separate group than `/api/cluster` in Swagger UI?

File: http/src/main/resources/swagger/api-docs.yaml [hunk 7]
```diff
@@ -61,6 +63,62 @@ paths:
               schema:
                 $ref: "#/components/schemas/LeonardoVersion"
       security: []
+  /api/google/v1/runtime/{googleProject}/{name}:
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml:66]: Once https://github.com/DataBiosphere/leonardo/pull/1321 is merged let's add the other routes

File: http/src/main/resources/swagger/api-docs.yaml [hunk 8]
```diff
@@ -1393,6 +1451,95 @@ components:
         customClusterEnvironmentVariables:
           type: object
           description: Optional environment variables to be set on the cluster.
+    CreateRuntimeRequest:
+      description: ""
+      type: object
+      properties:
+        labels:
+          type: object
+          description: The labels to be placed on the cluster. Of type Map[String,String]
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: s/cluster/runtime

File: http/src/main/resources/swagger/api-docs.yaml [hunk 9]
```diff
@@ -1393,6 +1451,95 @@ components:
...
+        userJupyterExtensionConfig:
+          $ref: "#/components/schemas/UserJupyterExtensionConfig"
+        jupyterExtensionUri:
+          type: string
+          description: Deprecated. Please use userJupyterExtensionConfig.
+        jupyterUserScriptUri:
+          type: string
+          description: >
+            Optional GCS object URI to a bash script the user wishes to run
+            inside their jupyter
```
Comment (rtitle) [inline http/src/main/resources/swagger/api-docs.yaml]: Seems odd this is double spaced.. did the formatter do that?

File: http/src/main/resources/swagger/api-docs.yaml (hunk 6 above)
Comment (Qi77Qi) [reply http/src/main/resources/swagger/api-docs.yaml:121]: I think it is in separate group (tag)

File: http/src/main/resources/swagger/api-docs.yaml (hunk 9 above)
Comment (Qi77Qi) [reply http/src/main/resources/swagger/api-docs.yaml]: weird. might be vsstudio

**Review by rtitle — APPROVED** [review]

**Pull request merged**